.github/workflows/build.yml

```diff
@@ -24,8 +26,12 @@ jobs:
           git config --local user.email "action@github.com"
           git config --local user.name "GitHub Action"
           git commit -m "Rebuilding documentation and generated files with Github Action" -a
-      - name: Push changes
-        uses: ad-m/github-push-action@master
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          branch: ${{ github.ref }}
+      # - name: Push changes
+      #   uses: ad-m/github-push-action@master
+      #   with:
+      #     github_token: ${{ secrets.GITHUB_TOKEN }}
+      #     branch: ${{ github.ref }}
+      - name: Run unit tests
+        run: |
+          make test
```

Comment:
what happens when the tests fail? does it make this command fail, and therefore the action fail?